modules/http-api/src/main/java/org/mule/service/http/api/HttpConstants.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.service.http.api;
+
+/**
+ * General purpose HTTP constants
+ */
+public abstract class HttpConstants {
+
+  public enum Methods {
+    POST, GET, PUT, OPTIONS, HEAD, DELETE;
+  }
+
+  public static enum Protocols {
+    HTTP("http", 80), HTTPS("https", 443);
+
+    private final String scheme;
+    private final int defaultPort;
+
+    Protocols(String scheme, int defaultPort) {
+      this.scheme = scheme;
+      this.defaultPort = defaultPort;
+    }
+
+    public String getScheme() {
+      return scheme;
+    }
+
+    public int getDefaultPort() {
+      return defaultPort;
+    }
+  }
+
+  public static final class RequestProperties {
```

Comment:
This should not go here but rather in the transport constants since they refer to the inbound/outbound properties handled by the HTTP module and transport, in fact they are probably only needed in the compatibility module.